### ISSUE TITLE
bolt upload throws if source or destination have trailing slashes

### ISSUE THREAD

**Issue by rogfut** [puppetlabs/bolt#1914]
## Describe the Bug
when using bolt file upload in powershell, if the source or destination has a trailing backslash in the path, bolt throws:

`Command requires a targeting option: '--targets', '--rerun', or '--query'`

## Expected Behavior
I would expect bolt to either handle the trailing slash, or to provide a more informative error.  The current error is misleading.

For comparison, the trailing backslash is supported in powershell's copy-item

## Steps to Reproduce
1. Open Powershell
2. use bolt file upload and include a trailing slash in either the source or destination path or both:

`bolt file upload c:\temp\dir1\ c:\temp\dir2\ --targets server01`

3. The error is thrown:

`Command requires a targeting option: '--targets', '--rerun', or '--query'`

## Environment
- Bolt Version 2.8.0
- PowerShell 5.1 build 17763 revision 1007
- Windows Version 10.0.17763

## Additional Context
In the example above I didn't include --transport -u or -p because it's defined in my inventory.yaml, however if I pass these params the error throws before the password prompt occurs which indicates to me the error is throwing before the winrm connection is made.  For this reason, --debug doesn't work to get any extra information about the failure.

**Comment by lucywyman** [comment]
Huh, I just tried the same on Powershell 5.1 (Windows 2019, Bolt 2.16.0) and it seems to have worked:
```
PS C:\Users\Administrator> bolt file upload C:\cygwin64\home\Administrator\.ssh\ C:\Users\Administrator\ -t localhost   Started on localhost...                                                                                                 Finished on localhost:                                                                                                    Uploaded 'C:\cygwin64\home\Administrator\.ssh\' to 'localhost:C:\Users\Administrator\'                                Successful on 1 target: localhost                                                                                       Ran on 1 target in 0.04 sec                                                                                             PS C:\Users\Administrator> $PSVersionTable                                                                                                                                                                                                      Name                           Value                                                                                    ----                           -----                                                                                    PSVersion                      5.1.17763.771                                                                            PSEdition                      Desktop                                                                                  PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                  BuildVersion                   10.0.17763.771                                                                           CLRVersion                     4.0.30319.42000                                                                          WSManStackVersion              3.0                                                                                      PSRemotingProtocolVersion      2.3                                                                                      SerializationVersion           1.1.0.1                                                                                                                                                                                                                                                                                                                                  PS C:\Users\Administrator>  
```
And the file is actually there. We did just modify how file upload works, which may have fixed this issue - if you upgrade to Bolt 2.16.0 does the issue persist?

**Comment by rogfut** [comment]
@lucywyman This is now working for me after upgrading to Bolt 2.16.0.  My windows version & ps version remain the same on my localhost and my remotes.  Thanks!
